mealie/core/security/providers/ldap_provider.py

```diff
@@ -0,0 +1,179 @@
+from datetime import timedelta
+
+import ldap
+from fastapi import Request
+from ldap.ldapobject import LDAPObject
+from sqlalchemy.orm.session import Session
+
+from mealie.core import root_logger
+from mealie.core.config import get_app_settings
+from mealie.core.security.providers.credentials_provider import CredentialsProvider
+from mealie.db.models.users.users import AuthMethod
+from mealie.repos.all_repositories import get_repositories
+from mealie.schema.user.user import PrivateUser
+
+logger = root_logger.get_logger("ldap_provider")
```

Comment:
Can we initialize this within the class?

Reply:
Absolutely, done!